feat: 용병 구인글 작성 페이지 마크업 및 기능(기능은 세부설명 미완성) (MAT-116)

HongJungKim-dev opened the pull request:
## 추가한 기능

## 체크리스트
- [x] 세부설명 기능 (textarea를 div나 input 으로 개선, handle함수 작성)
- [ ] 매칭 글작성과 공통되는 부분 컴포넌트화 (장소, 날짜, 포지션, 종목 - 필터, 연령)
- [x] 날짜 오늘로부터 한달만 가능하도록 제한
- [x] 예외처리 -> 날짜 미입력시, 시간 경기시작이 끝보다 느린경우, 시간 미입력, 장소 미입력
- [x] 날짜, 시간, 장소 제외 값들은 default 값으로 서버로 전송 가능 
라이브러리
* 날짜 선택 - [[링크](https://mui.com/components/date-picker/)]
* 시간 선택 - [[링크](https://mui.com/components/time-picker/)]

## 구현 결과
![image](https://user-images.githubusercontent.com/58525009/146351525-2afa70cb-449a-4b59-a398-f59557ef0e97.png)
* 날짜선택
![image](https://user-images.githubusercontent.com/58525009/145557521-24959415-ad9d-4780-847c-15a7a8f0a10e.png)

* 시간선택
![image](https://user-images.githubusercontent.com/58525009/146351573-0724d011-cd6e-4bbc-a5e6-4d91c52e9354.png)

HongJungKim-dev committed:
feat: 용병 구인글 작성 페이지 마크업 및 기능(기능은 세부설명 미완성)

94chl committed:
feat: input, inputCheckBox, inputDetail 컴포넌트 구현(MAT-94) (#9)

* feat: header, tabBar 컴포넌트 추가(MAT-42)

* feat: header, tabBar 컴포넌트 추가

* docs: 헤더 css 수정

* docs: lint 적용

* chore: fontawesome 관련 패키지 제거

* refactor: scss 클래스명 구조분해할당으로 변경

* chore: 각 컴포넌트의 index  파일명 변경(코드컨벤션)

* feat: 범용적 사용이 가능한 CustomHooks 구현(MAT-72)

* feat: team-sub-page 구현(MAT-51)

* feat: CustomInput 구현

* feat: CustomLabel 구현

* feat: CustomButton 구현

* feat: 팀 생성 페이지 마크업 구현

* refactor: 불필요한 console.log 제거

* feat: TeamInfoEdit 페이지 마크업

* refactor: common 파일 import 방식 변경

* refactor: AGE_GROUP & SPORT_CATEGORY 상수화

* refactor: 불필요한 TeamCreate 컴포넌트 제거

* refactor: 불필요한 TeamCreate 컴포넌트 제거

* refactor: 불필요한 onClick 메소드 제거

* chore: 추적되지 않은 파일 업로드

* refactor: useFor범용적으로 사용할  타입 추가

* feat: 팀 생성 및 수정 페이지에 사용되는 validation util 추가

* feat: 팀 생성 및 수정 페이지에 validation 추가

* style: 가독성 향상을 위한 개행 추가

* docs: PR Template 추가

PR Template 추가

* feat: Router 컴포넌트 구현 및 라우팅 구현 (MAT-86) (#6)

* feat: router const 추가

router path 관련한 라우팅 상수 추가

* feat: AuthorizedRoute 컴포넌트 추가

인증이 필요한 컴포넌트의 동적 라우팅을 위한 AuthorizedRoute 컴포넌트 추가

* feat: Router 컴포넌트 추가

라우팅과 관련된 로직을 처리하는 Router 컴포넌트 추가

* feat: App.tsx에Router 컴포넌트 렌더링

Router 컴포넌트 렌더링으로 App에 라우팅 기능 구현

* feat: 상수 이름 추가 및 수정

MERCENARY -> HIRES로 수정
MATCHING -> MATCHES로 수정
SETTING 추가

* feat: 상수명 변경 및 SETTING 추가

MATCHING -> MATCHES
MERCENARY -> HIRES

* feat: 구조분해할당 추가 및 명확한 props 작성

구조분해할당 추가 및 exact, path props 추가했습니다

* feat: input 컴포넌트 3가지 구현

* refactor: 변수명 피드백 반영

* refactor: form태그 제거 및 이에 따른 타입, 이벤트, 클래스 변경

* style: 스타일린트 적용

* refactor: input 스타일 props추가, 아이콘 props 삭제

* refactor: inputCheckBox 스타일 props추가

* style: 미사용 scss 제거

* style: 미적용 scss 수정

* style: 미적용 scss 수정

* refactor: inputDetail 미완성 및 낮은 활용도로 삭제

* refactor: 체크안된 박스 관련 스타일 변수명을 unCheckedBox~로 변경

* style: input 미적용 css 수정

* refactor: input 높이 지정 파라미터추가

* feat: inputDetail 컴포넌트(contentEditable) 추가

Co-authored-by: DongHyeon Kim <49136186+Kimbangg@users.noreply.github.com>
Co-authored-by: 오재원 <41986911+ProgWon@users.noreply.github.com>

94chl commented on package.json:
```diff
@@ -13,9 +13,15 @@
     "@reduxjs/toolkit": "^1.6.2",
     "axios": "^0.24.0",
     "classnames": "^2.3.1",
+    "date-fns": "2.27.0",
```
라이브러리 추가 조심스럽네요

94chl commented on src/pages/Hires/HiresWritePage.tsx:
```diff
@@ -0,0 +1,275 @@
+  };
+
+  return (
+    <>
```
저희 Fragment 쓰기로 했던것 같습니다

Kimbangg committed:
feat: 팀원 상세보기 페이지 구현(MAT-103) (#13)

* feat: 팀원 상세보기 페이지 구현(with 가이드 주석)

* feat: Props에 따라 용병 | 팀원 정보를 렌더링하는 List Component 생성

* refactor: 하드 코딩 -> List 컴포넌트를 이용해 렌더하는 방식으로 변경

* feat: MemberList Nested 구조 개선을 위해 MemberListElement 컴포넌트 추가

* refactor: MemberListElement를 통한 코드 리팩토링

* refactor: 네이밍 정정 & MemberList 적용

* feat: Member들의 grade를 변경할 수 있는 로직 추가

* refactor: 여러 개의 GraderMemberListElement를 MemberListElem 컴포넌트 추상화

* refactor: 불필요한 테스트 버튼 제거

* refactor: onClick 권한을 컨테이너가 통제할 수 있도록 li -> div 태그로 변경

* feat: 삭제할 멤버(checkbox) 클릭 시, deletedMember 배열에 push되는 로직 추가

* feat: 멤버 권한 관련 API 로직 추가

* feat: 용병 페이지에서는 위임 버튼을 사라지게 하는 로직 추가

* refactor: 불필요한 console.log 제거

* refactor: 불필요한 fragment 제거

* refactor: type | undefined -> typeName? 로 변경

* fix: props를 받지 않아서 발생하는 에러 수정

Kimbangg committed:
feat: 팀 상세보기 페이지 (MAT-41) (#12)

* refactor: Response데이터 형식에 맞게 API Core 로직 수정

* feat: 에러처리 로직 추가

* refactor: export 방식 변경( -> index.ts 모아 내보내기)

* feat: 팀 상세보기 페이지 구현

* refactor: 네이밍의 일관성을 위해 tagType -> tagName으로 변경

* feat: 권한에 따라 회원탈퇴 또는 팀탈퇴 버튼이 생기도록 구현

* feat: 팀 페이지 뷰에 용병 리스트 추가

* feat: 용병 & 멤버 리스트 더보기(Link)  추가

* refactor: Modal 추가 이전에, 대안적으로 confirm 함수 사용

* refactor: useState의 타입을 명시적으로 변경

* refactor: import 방식을 상대 ->  절대경로로 수정

* refactor: 주장 클릭 시, 메세지 또는 개인정보에 접근할 수 있도록 id 로 변경

Kimbangg committed:
refactor: 머지 이전에 발견 못했던 코드리뷰 반영 (#18)

* refactor: window.location -> useParams (MAT-41, 미반영된 코멘트 반영)

* refactor: Event Type을 명시적으로 변경(MAT-103)

* fix: form 태그 바깥에 버튼을 둠으로써, event 발생이 안되는 에러 정정

* refactor: useParams로 path variable 받을 수 있도록 리팩토링

* refactor: Router Url 주소 변경

Kimbangg committed:
feat: 매치 상세페이지 구현 (MAT-152) (#17)

* chore: 당근마켓 keyframe/tabs 라이브러리 추가

* feat: MatchDetail 컴포넌트에 기본 스타일 & keyframe/tabs 라이브러리 적용

* feat: Common Color Scss 파일 추가

* refactor: css 컨벤션에 맞게 네이밍 변경

* refactor: Web Title 변경(title -> MatchUs)

* refactor: key props 타입 선언

* feat: MatchListElement Component 구현

* feat: 매치 상세보기 페이지 구현

* feat: MatchListElement Component에 매칭 시간 추가

94chl committed:
refactor: 헤더, 탭바 로직 통일 및 스타일린트 적용 (#15)

* feat: header, tabBar 컴포넌트 추가

* docs: 헤더 css 수정

* style: lint 적용

* chore: fontawesome 관련 패키지 제거

* refactor: scss 클래스명 구조분해할당으로 변경

* refactor: 각 컴포넌트의 index  파일명 변경(코드컨벤션)

* refactor: 헤더 Link -> useHistory 변경, styles 구조분해할당 함수밖으로 꺼냄

* style: 스타일린트 적용

* refactor: App.tsx에 DefaultTemplate 추가

* refactor: 라우터 추가로 DefaultTempalte의 BrowserRouter 삭제

* refactor: input 드랍박스의 기본값을 옵션들중 제일 첫번째 값으로  설정

94chl commented on src/pages/Hires/HiresWritePage.tsx:
```diff
@@ -0,0 +1,275 @@
+  const [formatedEndTime, setFormatedEndTime] = useState<string>('');
+  const [maximumDate, setMaximumDateDate] = useState<any>(new Date());
+  const [currentDate, setCurrentDate] = useState<any>(new Date());
+  const [currentMonth, setCurrentMonth] = useState<any>(new Date().getMonth());
```
any 타입 특정해주는게 좋지 않을까요

94chl commented on src/pages/Hires/HiresWritePage.tsx:
```diff
@@ -0,0 +1,275 @@
+    setFormatedEndTime(formettedEndTime);
+  }, []);
+
+  useEffect(() => {
```
useEffect가 두개로 나뉘었어요

94chl commented on src/pages/Hires/HiresWritePage.tsx:
```diff
@@ -0,0 +1,275 @@
+    setRegion(value);
+  };
+
+  const handleChangeGroundName = (event: React.ChangeEvent<HTMLSelectElement>) => {
```
단순히 값을 받아와서 setState만 한다면 하나의 함수로 통일할수 있을것 같아요

Kimbangg committed:
refactor: axios return data를 response.data.data로 변경 (#21)

Kimbangg committed:
feat: not found page(MAT-170) (#25)

* feat: 기본 색상 추가

* chore: ErrorIcon 추가

* feat: NotFound 페이지 생성

* feat: 스크린 리더에 읽히는 페이지 정보 추가

* feat: 스크린 리더에 읽히는 페이지 정보 추가

* refactor: 새로고침 버튼을 반응형으로 변경

HongJungKim-dev committed:
feat: route path를 적용하여 컴포넌트 라우팅 (#27)

94chl committed:
feat: Match 관련 페이지 라우터 연결 (#28)

HongJungKim-dev committed:
feat: 용병(및 매치) 리스트 페이지 마크업 작성 (#5)

* feat: 용병(및 매치) 리스트 페이지 마크업 작성

* style: 외부 라이르러리와 내부파일 경로 개행으로 구분

* refactor: 백엔드 API 설계에 맞게 매너온도 데이터 타입 수정

* feat: 로그인이 안된 경우, 권한이 없는 경우에는 용병 매칭 추가 버튼 비활성화

* refactore: 용병, 매칭 리스트의 key를 상위컴포넌트에 이동

* refactor: 모듈로 scss를 관리하기 때문에 BEM방식 제거

* refactor: 파일 경로를 절대경로로 수정

* refactor: 주석 일괄적용으로 수정

* docs: 추후 구현할 사항을 주석으로 작성

Kimbangg committed:
refactor: API Core를 백엔드 환경에 맞게 정정 (#26)

Kimbangg committed:
feat: 팀 페이지 라우터 주소 추가 (#30)

Kimbangg committed:
feat: 팀 하위 페이지에 API 연결(MAT-163) (#20)

* refactor: 백엔드 환경에 맞게 api core 수정

* feat: 팀 생성 API 연동로직 추가

* refactor: 입력과 동시에 validation이 되도록 로직 변경

* feat: 팀 이름 중복확인 API  추가 및 적용

* refactor: 각 컴포넌트에서 쓰이던 API 요청을 통합 관리

* refactor: 범용적인 사용을 위한 hasCategoryTitle Props 추가

* refactor: 범용적인 컴포넌트 적용 및 이벤트 핸들러 개선(useCallback)

* refactor: 반복되는 type -> types로 폴더에서 통합관리

* refactor: 반복되는 type -> types로 폴더에서 통합관리

* refactor: 반복되는 type Import해서 사용

* feat: 팀원/용병/매칭 정보가 없을 때 렌더링되는 뷰를 추가

* feat: API 연결 & 데이터가 비었을 때의 뷰 추가

* refactor: axios return data를 response.data.data로 변경

* refactor: axios return data에 맞춰서 데이터를 받아오도록 변경

* feat: 팀 수정 API 연동

* refactor: isDisabled의 값에 결정하는 요소로 isLoading 추가

* feat: 팀 생성 및 수정 시, 사진이 없으면 버튼 비활성화가 되는 로직 추가

* refactor: array.filter 로직을 명시적으로 수정

94chl committed:
feat: Match 페이지 구현(MAT-35) (#10)

* feat: API Core(axios) 로직 추가

* feat: matchInfo컴포넌트(매치정보) 추가

* feat: matchDetail컴포넌트(매치상세정보) 추가

* feat: TeamCard컴포넌트(팀정보) 추가

* feat: TeamButton컴포넌트(매칭버튼) 추가 및 컴포넌트 export

* feat: Match페이지 추가

* refactor: Match페이지 상대팀 존재 시 상대팀 정보 표시 적용

* refactor: url 주소 변경(matching -> matches)

* refactor: 용병, 매칭글을 posts로 통합->리덕스 대거 수정

* refactor: 리덕스 수정에 따른 컴포넌트 경로 변경

* feat: posts컴포넌트(게시글 목록) 추가

* feat: postItem컴포넌트(각 게시글 카드) 추가

* feat: MatchesPage페이지(매칭 리스트) 추가 및 페이지 코드 컨벤션 준수

* refactor: 포스트 리덕스에서 매칭수락 관련 로직 제거

* refactor: 매칭 상세보기 리덕스 액션함수 수정

* docs: 스타일린트 적용

* docs: 스타일린트 적용

* docs: 미사용 컴포넌트 제거

* feat: MatchApplyModal(매칭 신청) 기능 추가

* feat: MatchApproveModal(매칭 수락) 기능 추가

* feat: MatchReviewModal(매칭 평가) 기능 추가

* feat: 매칭 신청, 수락, 평가 컴포넌트 내보내기 추가

* feat: 매칭 리덕스 추가

* feat: 매칭 신청, 수락, 평가 모달 추가

* refactor: 매칭 신청, 수락, 평가 모달 제어를 위한 버튼 기능 추가

* feat: header, tabBar 컴포넌트 추가(MAT-42)

* feat: header, tabBar 컴포넌트 추가

* docs: 헤더 css 수정

* docs: lint 적용

* chore: fontawesome 관련 패키지 제거

* refactor: scss 클래스명 구조분해할당으로 변경

* chore: 각 컴포넌트의 index  파일명 변경(코드컨벤션)

* feat: 범용적 사용이 가능한 CustomHooks 구현(MAT-72)

* feat: team-sub-page 구현(MAT-51)

* feat: CustomInput 구현

* feat: CustomLabel 구현

* feat: CustomButton 구현

* feat: 팀 생성 페이지 마크업 구현

* refactor: 불필요한 console.log 제거

* feat: TeamInfoEdit 페이지 마크업

* refactor: common 파일 import 방식 변경

* refactor: AGE_GROUP & SPORT_CATEGORY 상수화

* refactor: 불필요한 TeamCreate 컴포넌트 제거

* refactor: 불필요한 TeamCreate 컴포넌트 제거

* refactor: 불필요한 onClick 메소드 제거

* chore: 추적되지 않은 파일 업로드

* refactor: useFor범용적으로 사용할  타입 추가

* feat: 팀 생성 및 수정 페이지에 사용되는 validation util 추가

* feat: 팀 생성 및 수정 페이지에 validation 추가

* style: 가독성 향상을 위한 개행 추가

* docs: PR Template 추가

PR Template 추가

* feat: Router 컴포넌트 구현 및 라우팅 구현 (MAT-86) (#6)

* feat: router const 추가

router path 관련한 라우팅 상수 추가

* feat: AuthorizedRoute 컴포넌트 추가

인증이 필요한 컴포넌트의 동적 라우팅을 위한 AuthorizedRoute 컴포넌트 추가

* feat: Router 컴포넌트 추가

라우팅과 관련된 로직을 처리하는 Router 컴포넌트 추가

* feat: App.tsx에Router 컴포넌트 렌더링

Router 컴포넌트 렌더링으로 App에 라우팅 기능 구현

* feat: 상수 이름 추가 및 수정

MERCENARY -> HIRES로 수정
MATCHING -> MATCHES로 수정
SETTING 추가

* feat: 상수명 변경 및 SETTING 추가

MATCHING -> MATCHES
MERCENARY -> HIRES

* feat: 구조분해할당 추가 및 명확한 props 작성

구조분해할당 추가 및 exact, path props 추가했습니다

* refactor: 페이지index, 컴포넌트 index 수정

* refactor: post 마크업 수정 및 컴포넌트 export

* style: post 스타일 추가

* style: css 수정

* refactor: path variable 로직 useParams로 변경

* refactor: match 리덕스 네이밍 변경

* refactor: dummy 데이터 업데이트

* refactor: e:any 타입 변경 및 matchId 로직 변경

* refactor: matchId 리덕스에서 관리하게 변경

* refactor: 테스트용 콘솔로그 삭제

* feat: 상수(평가 태그, 종목별 제한 선수) 추가

* refactor: 선수 표시 기준, 선택 평가 제한을 명시적이게 상수로 변경

* refactor: selectedTeam 타입 지정

* refactor: 게시글 등록 로직 변경(추후 사용자 권한 체크 로직 추가 필요)

* refactor: 불필요한 key 제거

* refactor: teamCard 반복문 key값 추가

* refactor: input 기본커서 auto로 변경

* refactor: api 변경에 따른 더미데이터 변경

* refactor: api 변경에 따른 스토어 변경

* refacotr: api 변경에 따른 페이지 수정

* refactor: api 변경에 따른 컴포넌트 수정

* refactor: input 컴포넌트 기능 수정

* feat: Matches, Match  라우터 연결

* refactor: MatchId 스토어 사용

* refactor: css px -> rem으로 변경

* refactor: input 기능 수정

* refactor: api 데이터 형식 변경으로 인한 변수, 데이터 수정

* refactor: posts 스토어에서 매치 관련 분리작업 및 posts 스토어 삭제

* refactor: Match 관련 types 분리

* style: eslint 적용

* refactor: 태그 로직 명시적으로 변경

Co-authored-by: DongHyeon Kim <49136186+Kimbangg@users.noreply.github.com>
Co-authored-by: 오재원 <41986911+ProgWon@users.noreply.github.com>

94chl committed:
feat: 매칭 등록 페이지 추가 (MAT-111) (#19)

* feat: NewMatch(매칭 등록) 페이지 추가

* feat: input 컴포넌트 export

* feat: Match에 필요한 상수 추가

* feat: Match에 필요한 Match 스토어 추가

* feat: 날짜, 시간 input 패키지 추가

* feat: 더미 데이터 추가

* refactor: api 데이터 변경으로 인한 더미 데이터 변경

* refactor: api 변경에 따른 페이지 수정

* feat: 라우터 연결

* refactor: input 컴포넌트 기능 수정

* style: 기능별 코드 재정렬(useEffect 종합 등)

* refactor: cost 숫자타입 지정 관련 오류 수정

* refactor: 상수 데이터 수정

* refactor: input 기능 수정(props로 value추가)

* refactor: api 데이터 형식 변경으로 인한 변수, 데이터 수정

* refactor: 시간, 날짜 에러 세분화, 함수 이름 명료화

* refactor: 라우터 url 변경

HongJungKim-dev committed:
refactor: 용병페이지 라우터 주소 추가 재PR (MAT-173) (#31)

* feat: route path를 적용하여 컴포넌트 라우팅

* refactor: 기존 용병 페이지를 route path에 연결한 컴포넌트로 이동

* refactor: merge과정에서 실수로 경로 잘못 적은것 수정

94chl committed:
feat: 매칭 수정 페이지 추가(MAT-112) (#23)

* feat:EditMatchPage 추가

* feat: 수정페이지 라우터 연결

* feat: 수정페이지 관련 스토어, 더미데이터 추가

* refactor: 수정페이지 관련 inputs 컴포넌트 최신화

* feat: 날짜, 시간 패키지 추가

* refactor: api 데이터 형식 변경으로 인한 변수, 데이터 수정

* refactor: 시간, 날짜 에러 세분화, 함수 이름 명료화

* refactor: 용병, 매칭 분리로 스토어에서 매칭관련 로직 분리

* refactor: 달력이 저장된 날짜 불러오지 못하던 오류 수정

ProgWon committed:
feat: 회원가입 페이지 기능 구현 (MAT-99) (#22)

* feat: 회원가입 validation 기능 구현

signupValidation.ts 구현

* feat: 회원가입 페이지 기능 구현

API 연동을 제외한 회원가입 페이지 기능 구현

* feat: 회원가입 페이지 공통 컴포넌트 ValidateInput.tsx 구현

ValidateInput.tsx 컴포넌트 구현

* feat: 회원가입 상수 추가

회원가입에 필요한 상수 추가

* feat: 회원가입에 필요한 API 구현

회원가입에 필요한 API 구현

* feat: authSlice에 login 기능 구현

login 기능 구현

* feat: 인증과 관련된 타입 정의

인증과 관련된 타입 정의

* feat: 회원가입에 필요한 검증 폼 구현

회원가입에 필요한 검증 실패 메세지 구현 및 성공 메세지 구현

* feat: 회원가입 페이지 구현 (API 미연동)

회원가입 API 연동을 제외한 회원가입 페이지 기능 구현

HongJungKim-dev committed:
feat: 용병 상세페이지 (MAT-57) (#14)

* feat: 용병 상세페이지 마크업

* chore: 불필요한 내용 삭제

* chore: 불필요한 Fragment 삭제

* refactor: 관심사가 모이도록 컴포넌트 수정

* refactor: 백엔드 API 설계에 맞게 데이터 타입 수정

Kimbangg committed:
feat: 팀 페이지 라우팅 연결 (MAT-195) (#33)

* feat: 경기 등록 Link에 MATCH_POST 페이지 연결

* feat: 팀 상세보기 페이지 라우터 연결

* refactor: hires pages import 방식을 상대 ->  절대 경로로 변경

* feat: NotFound 페이지 라우터에 추가

ProgWon committed:
feat: 로그인 페이지 추가(MAT-50) (#29)

* feat: 회원가입 validation 기능 구현

signupValidation.ts 구현

* feat: 회원가입 페이지 기능 구현

API 연동을 제외한 회원가입 페이지 기능 구현

* feat: 회원가입 페이지 공통 컴포넌트 ValidateInput.tsx 구현

ValidateInput.tsx 컴포넌트 구현

* feat: 회원가입 상수 추가

회원가입에 필요한 상수 추가

* feat: 회원가입에 필요한 API 구현

회원가입에 필요한 API 구현

* feat: authSlice에 login 기능 구현

login 기능 구현

* feat: 인증과 관련된 타입 정의

인증과 관련된 타입 정의

* feat: 회원가입에 필요한 검증 폼 구현

회원가입에 필요한 검증 실패 메세지 구현 및 성공 메세지 구현

* feat: 회원가입 페이지 구현 (API 미연동)

회원가입 API 연동을 제외한 회원가입 페이지 기능 구현

* feat: 회원가입 API 연동 및 에러처리

회원가입 API 연동 및 검증 안된 폼에서 중복처리 호출하던 에러 수정

* feat: 재인증 API 구현

requestReAuth API call 구현

* feat: 토큰 만료 처리를 위한 request interceptor 구현

토큰 만료시 login 페이지로 리다이렉트 하도록 interceptor 구현

* feat: 로그인 페이지 구현

로그인 페이지 구현

* feat: 토큰 만료 상태, 타입 추가 및 로직 구현

토큰 만료 시간 상태 관리 및 타입 추가후 로직 구현

* feat: store에 auth Reducer 추가

auth Reducer 추가

* refactor: auth -> auths 로 수정

파일명 수정

* refactor: handler 함수명에 handle 추가 및 경로 수정

handler 함수인 경우 handle + 메소드로 함수명 수정
절대 경로 + 모아 보내기로 수정

* refactor: auth -> auths로 경로명 변경 및  같은 type의 경우 extend 추가

auth -> auths로 auth api의 경로명 변경
auths의 type중 내용이 같은 경우 extends 사용

* refactor: interface 통합

* refactor: 오타수정

Kimbangg committed:
feat: 팀원 관리 API 연결 (MAT-196) (#34)

* refactor: 더미 데이터 삭제 & 팀 멤버 정보 API 연동

* refactor: 특정 멤버가 없을 때 렌더링을 안하는 로직 추가

Kimbangg committed:
feat: 공통 모달 컴포넌트 (MAT-171) (#32)

* feat: Modal 구현을 위한 Portal 구현

* feat: Portal을 이용한 Modal Common Component 구현

* feat: Portal 적용을 위한 modal-root div 추가

* feat: 범용적으로 사용이 가능한 CustomModalDialog Component 구현

Kimbangg committed:
feat: 팀 관련 API 연동 (MAT-196) (#37)

* refactor: authSlice에서 발생하는 에러 수정

* refactor: 버튼 라벨을 받을 수 있도록 모달 수정

* feat: 팀원 초대 API 연결

* refactor: 팀원 Grade 네이밍을 백엔드에 맞춰 변경

* refactor: 팀원 권한 변경 API 연동

* refactor: 팀원 방출 API 연동

* refactor: 팀원 방출 API 연동

* refactor: 반복 사용되는 MemberElementType Interface를 타입으로 분리

* refactor: MemberElementType를 Import 해서 사용하도록 변경

* refactor: 전체 멤버 조회 API로 수정

* refactor: 백엔드 환경에 맞게 객체 이름을 변경(MatchSummary -> teamMatches)

* refactor: tsConfig 일부 수정

* refactor: 팀 상세 페이지에서 보이는 팀원 & 용병 수를 제한하는 로직 추가

HongJungKim-dev committed:
Merge branch 'feature/hires-write(MAT-116)' of https://github.com/prgrms-web-devcourse/Team_CherryBoys_MatchUs_FE into feature/hires-write(MAT-116)

HongJungKim-dev commented:
컨플릭트 해결하고 커밋 안하고 계속 작업해서 그때 다시 커밋메시지 작성하려니깐 Merge자동메시지 나오길래 conflict해결 커밋이랑 작업 커밋 나눠서 작성하려고 커밋메시지 저장안하고 나가니깐 이렇게 되었어요. 원래 저장안하고 나가면 커밋 실패하는것으로 알고있어서 그렇게 한건데... 그냥 PR Close하고 다시 PR하려고 하는데 어떻게 생각하시나요?
git pull해도 already updated가 뜹니다.
